orchagent/portsorch.cpp

```diff
@@ -5666,6 +5764,20 @@ void PortsOrch::updatePortOperStatus(Port &port, sai_port_oper_status_t status)
     if (port.m_type == Port::PHY)
     {
         updateDbPortOperStatus(port, status);
+        if (port.m_autoneg > 0)
+        {
+            if (status == SAI_PORT_OPER_STATUS_UP)
+            {
+                updatePortStateAutoNeg(port);
+                m_port_state_poll[port.m_alias] &= ~PORT_STATE_POLL_AN;
```

Comment:
this is an async notification from low level, how could it make sure that the adv speeds has been retrieved already?

Reply:
This is a case of a link-up event, hence the remote advertisement is already received and processed by the ASIC, and the remote advertisement is surely available at this moment.